raiden/transfer/node.py

```diff
@@ -482,7 +482,8 @@ def handle_delivered(chain_state: ChainState, state_change: ReceiveDelivered) ->
             filtered_queue = [
                 message
                 for message in queue
-                if message.message_identifier != state_change.message_identifier
+                if message.message_identifier != state_change.message_identifier and
```

Comment:
I think it'd be an `or` here, as you want to filter out messages `if not (same message_id and same sender)` =~ `if not same message_id or not same sender`